### ISSUE TITLE
Filter by relationship attribute

### ISSUE THREAD

**Issue by brunocascio** [marcelgwerder/laravel-api-handler#10]
/user/dogs?breed.name=labrador 

is it possible?

In this case:

dogs->belongsTo('Breed');

In breed, name is an attribute.

**Comment by sptdigital** [comment]
For the issue with dot's in the name, it is worth reading [this thread](https://stackoverflow.com/a/8763572).

Has there been any changes to the way relationships are handled in Eloquent, or are they still done in a similar way?

**Comment by sptdigital** [comment]
Having created a workaround for the aforementioned issue, the final query is missing the join for any tables defined in the '_with'.

i.e.
my.api/someobject/?_with=contract&contract.contractnumber-lk=1234 will report something along the lines of 'unknown field contract.contractnumber on table someobject'.

The _with is getting parsed, but is not getting used.

**Comment by Pederytter** [comment]
Yeah I'm having the same problem. A patch for this to work would be awesome.

**Comment by sptdigital** [comment]
Here's how I got around this issue ....

`if(Input::has('tender_type')) {
            $relatedModelsField= Input::get('relatedModelsField');
            $model = Model::whereHas('relatedModel', function ($query) use ($relatedModelsField) {
                $query->where('relatedModelsField', '=', $relatedModelsField);
            });
        } else {
            $model= new Model();
        }

        $builder = ApiHandler::parseMultiple($model, array(), Input::except('relatedModelsField'))->getBuilder();`

**Comment by jadonbrownza** [comment]
Would love to see this
